pkg/endpoint/endpoint.go

```diff
@@ -1580,16 +1580,13 @@ func (e *Endpoint) LeaveLocked(owner Owner) int {
 	}
 
 	if e.SecurityIdentity != nil {
-		lastUse, err := e.SecurityIdentity.Release()
+		err := e.SecurityIdentity.Release()
 		if err != nil {
 			log.WithError(err).WithField(logfields.Identity, e.SecurityIdentity.ID).
 				Error("Unable to release identity of endpoint")
 			errors++
 		}
-		if lastUse {
-			owner.RemoveNetworkPolicy(e.SecurityIdentity.ID)
-		}
-
+		owner.RemoveNetworkPolicy(e.IPv6, e.IPv4)
```

Comment:
Do we need to keep this in the `if e.SecurityIdentity != nil {` block? Would it be safe to move it before?

Reply:
I think that just means if you add an endpoint then delete it before we resolve the identity for it, then we'll make extraneous `RemoveNetworkPolicy()` calls before a NetworkPolicy is generated.